psi4/driver/driver.py

```diff
@@ -873,6 +873,158 @@ def properties(*args, **kwargs):
         return core.variable('CURRENT ENERGY')
 
 
+
+def optimize_geometric(name, **kwargs):
+
+    import geometric
+    import qcelemental as qcel
+
+    class Psi4NativeEngine(geometric.engine.Engine):
+        """
+        Run a Psi4 energy and gradient calculation for geometric without any I/O
+        """
+        def __init__(self, p4_name, p4_mol, **p4_kwargs):
+    
+            self.p4_name = p4_name
+            self.p4_mol = p4_mol
+            self.p4_kwargs = p4_kwargs
+    
+            molecule = geometric.molecule.Molecule()
+            molecule.elem = [p4_mol.symbol(i) for i in range(p4_mol.natom())]
+            molecule.xyzs = [p4_mol.geometry().np * qcel.constants.bohr2angstroms] # reshape?
+            molecule.build_bonds()
+                                 
+            super(Psi4NativeEngine, self).__init__(molecule)
+    
+        def calc(self, coords, dirname):
+            self.p4_mol.set_geometry(core.Matrix.from_array(coords.reshape(-1,3)))
+            self.p4_mol.update_geometry()
+            g, wfn = gradient(self.p4_name, return_wfn=True, molecule=self.p4_mol, **self.p4_kwargs)
```

Comment:
Regarding the fact that CBS calls apparently don't work: I didn't try it out, but I don't see why a `cbs` call with `cbs_metadata` in kwargs shouldn't work here... If anything is missing, let me know, maybe I can help.

Reply:
You're totally right. CBS might not have worked before DGAS's suggestion to use a custom engine, but now it works fine without any modification